src/modules/cpu/kernel/normalize.hpp

```diff
@@ -0,0 +1,340 @@
+                         Rpp32f *meanPtr, Rpp32f *invStdDevPtr, Rpp32f shift, Rpp32u *dims, Rpp32u *paramStride)
+{
+    Rpp32f *srcPtrTemp = srcPtr;
+    Rpp32f *dstPtrTemp = dstPtr;
+    Rpp32s paramIdx = 0;
+    for(Rpp32u i = 0; i < dims[0]; i++) {
+        Rpp32f *srcPtrTempRow = srcPtrTemp;
+        Rpp32f *dstPtrTempRow = dstPtrTemp;
+        for(Rpp32u j = 0; j < dims[1]; j++) {
+            *dstPtrTempRow++ = (*srcPtrTempRow++ - meanPtr[paramIdx]) * invStdDevPtr[paramIdx] + shift;
+            paramIdx += paramStride[0];
+        }
+        paramIdx = (!paramStride[1]) ? 0 : paramIdx + paramStride[1];
+        srcPtrTemp += (dstDescPtr->h > 1 and dstDescPtr->w > 1) ? srcDescPtr->strides.hStride : srcDescPtr->strides.wStride;
+        dstPtrTemp += (dstDescPtr->h > 1 and dstDescPtr->w > 1) ? dstDescPtr->strides.hStride : dstDescPtr->strides.wStride;
+    }
+}
+
+void normalize_2D_tensor_avx_axis2(Rpp32f *srcPtr, RpptDescPtr srcDescPtr, Rpp32f *dstPtr, RpptDescPtr dstDescPtr,
+                                   Rpp32f *meanPtr, Rpp32f *invStdDevPtr, Rpp32f shift, Rpp32u *dims, Rpp32u *paramStride)
+{
+    Rpp32f *srcPtrTemp = srcPtr;
+    Rpp32f *dstPtrTemp = dstPtr;
+    Rpp32s paramIdx = 0;
+    Rpp32u vectorIncrement = 8;
+    Rpp32u bufferLength = dims[1];
+    Rpp32u alignedLength = (bufferLength / 8) * 8;
+    Rpp32u numRows = dims[0];
+
+    __m256 pShift = _mm256_set1_ps(shift);
+    for(Rpp32u i = 0; i < numRows; i++)
+    {
+        Rpp32f *srcPtrTempRow = srcPtrTemp + i * srcDescPtr->strides.hStride;
+        Rpp32f *dstPtrTempRow = dstPtrTemp + i * dstDescPtr->strides.hStride;
+
+        // set mean and stddev
+        Rpp32f mean = meanPtr[i];
+        Rpp32f invStdDev = invStdDevPtr[i];
+        __m256 pMean, pInvStdDev;
+        pMean = _mm256_set1_ps(mean);
+        pInvStdDev = _mm256_set1_ps(invStdDev);
+
+        Rpp32u vectorLoopCount = 0;
+        for(; vectorLoopCount < alignedLength ; vectorLoopCount += 8)
+        {
+            __m256 pSrc = _mm256_loadu_ps(srcPtrTempRow);
+            __m256 pDst = _mm256_add_ps(_mm256_mul_ps(_mm256_sub_ps(pSrc, pMean), pInvStdDev), pShift);
+            _mm256_storeu_ps(dstPtrTempRow, pDst);
+            srcPtrTempRow += 8;
+            dstPtrTempRow += 8;
+        }
+        for(; vectorLoopCount < dims[1] ; vectorLoopCount += 8)
+             *dstPtrTempRow++ = (*srcPtrTempRow++ - mean) * invStdDev + shift;
+    }
+}
+
+RppStatus normalize_audio_host_tensor(Rpp32f* srcPtr,
+                                      RpptDescPtr srcDescPtr,
+                                      Rpp32f* dstPtr,
+                                      RpptDescPtr dstDescPtr,
+                                      Rpp32s *srcLengthTensor,
+                                      Rpp32s *channelsTensor,
+                                      Rpp32s axis_mask,
+                                      Rpp32f mean,
```

Comment:
We need to discuss on the mean and stddev parameters passed from rocAL
It should be a tensor from rocAL. If your passes a tensor of mean and tensor of stdDev the current code won't be able to handle it

Lets do this once all the other comments are handled